src/components/frame/LandingPage/hooks.ts

```diff
@@ -0,0 +1,125 @@
+import { useNavigate } from 'react-router-dom';
+
+import { useAppContext } from '~context/AppContext/AppContext.ts';
+import { useGetMembersCountQuery } from '~gql';
+import useBaseUrl from '~hooks/useBaseUrl.ts';
+import { CREATE_COLONY_ROUTE_BASE } from '~routes';
+import { getLastWallet } from '~utils/autoLogin.ts';
+
+export const useLandingPage = () => {
+  const {
+    canInteract,
+    joinedColonies,
+    joinedColoniesLoading,
+    user,
+    connectWallet,
+    wallet,
+    walletConnecting,
+    userLoading,
+  } = useAppContext();
+
+  const navigate = useNavigate();
+
+  const {
+    data: contributorsCount,
+    fetchMore,
+    loading: contributorsCountLoading,
+  } = useGetMembersCountQuery({
+    variables: {
+      filter: {
+        isWatching: { eq: true },
+        or: joinedColonies.map((colony) => ({
+          colonyAddress: { eq: colony.colonyAddress },
+        })),
+      },
+    },
+    onCompleted: (data) => {
+      if (data.searchColonyContributors?.nextToken) {
+        fetchMore({
+          variables: {
+            isWatching: { eq: true },
+            or: joinedColonies.map((colony) => ({
+              colonyAddress: { eq: colony.colonyAddress },
+            })),
+            nextToken: data.searchColonyContributors.nextToken,
+          },
+          updateQuery: (prev, { fetchMoreResult }) => {
+            if (!fetchMoreResult) return prev;
+
+            return {
+              ...prev,
+              getActionsByColony: {
+                ...prev.searchColonyContributors,
+                items: [
+                  ...(prev.searchColonyContributors?.aggregateItems[0]?.result
+                    ?.__typename === 'SearchableAggregateBucketResult'
+                    ? prev.searchColonyContributors?.aggregateItems[0]?.result
+                        ?.buckets ?? []
+                    : []),
+                  ...(fetchMoreResult.searchColonyContributors
+                    ?.aggregateItems[0]?.result?.__typename ===
+                  'SearchableAggregateBucketResult'
+                    ? fetchMoreResult.searchColonyContributors
+                        ?.aggregateItems[0]?.result?.buckets ?? []
+                    : []),
```

Comment:
Also here, is there any reason for needing to check the `__typename `?

Otherwise let's just refactor this part as below

Reply:
Same as above